python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -136,45 +142,100 @@ def cast_string_literal_to_string_view(context, builder, fromty, toty, val):
     return sv._getvalue()
 
 
-@cuda_lowering_registry.lower_cast(string_view, udf_string)
-def cast_string_view_to_udf_string(context, builder, fromty, toty, val):
+@cuda_lowering_registry.lower_cast(string_view, managed_udf_string)
+def cast_string_view_to_managed_udf_string(
+    context, builder, fromty, toty, val
+):
     sv_ptr = builder.alloca(default_manager[fromty].get_value_type())
-    udf_str_ptr = builder.alloca(default_manager[toty].get_value_type())
+
+    managed_ptr = builder.alloca(
+        default_manager[managed_udf_string].get_value_type()
+    )
+    udf_str_ptr = builder.gep(
+        managed_ptr, [ir.IntType(32)(0), ir.IntType(32)(1)]
+    )
+
     builder.store(val, sv_ptr)
     _ = context.compile_internal(
         builder,
         call_create_udf_string_from_string_view,
         nb_signature(types.void, _STR_VIEW_PTR, types.CPointer(udf_string)),
         (sv_ptr, udf_str_ptr),
     )
-    result = cgutils.create_struct_proxy(udf_string)(
-        context, builder, value=builder.load(udf_str_ptr)
-    )
-
-    return result._getvalue()
+    return _finalize_new_managed_udf_string(context, builder, managed_ptr)
 
 
-@cuda_lowering_registry.lower_cast(udf_string, string_view)
-def cast_udf_string_to_string_view(context, builder, fromty, toty, val):
-    udf_str_ptr = builder.alloca(default_manager[fromty].get_value_type())
+@cuda_lowering_registry.lower_cast(managed_udf_string, string_view)
+def cast_managed_udf_string_to_string_view(
+    context, builder, fromty, toty, val
+):
     sv_ptr = builder.alloca(default_manager[toty].get_value_type())
-    builder.store(val, udf_str_ptr)
+
+    managed_ptr = builder.alloca(default_manager[fromty].get_value_type())
+    builder.store(val, managed_ptr)
+
+    udf_str_ptr = builder.gep(
+        managed_ptr, [ir.IntType(32)(0), ir.IntType(32)(1)]
+    )
 
     context.compile_internal(
         builder,
         call_create_string_view_from_udf_string,
         nb_signature(types.void, _UDF_STRING_PTR, _STR_VIEW_PTR),
         (udf_str_ptr, sv_ptr),
     )
-
     result = cgutils.create_struct_proxy(string_view)(
         context, builder, value=builder.load(sv_ptr)
     )
-
     return result._getvalue()
 
 
-# utilities
+# Utilities
+_new_meminfo_from_udf_str = cuda.declare_device(
+    "meminfo_from_new_udf_str", types.voidptr(_UDF_STRING_PTR)
```

Comment:
It would be good to clear up whether this is a new meminfo from a UDF string, or a meminfo from a new UDF string, when thinking about resolving https://github.com/rapidsai/cudf/pull/18453/files#r2088557242